jina/peapods/pods/__init__.py

```diff
@@ -21,7 +21,27 @@
 
 
 class BasePod(ExitStack):
```

Comment:
This class can take way more common code from the CompoundPod and The BasePod. There is too much duplication.

Reply:
yes, the BasePod should have all the interface that the `Flow` needs to know to connect different `BasePods`

Reply:
I thought it might be more readable to have this separated and optimize it in a follow-up pr.
Now, I combined the common logic.